build/org.eclipse.pde.build/src/org/eclipse/pde/internal/build/site/PDEState.java

```diff
@@ -317,45 +317,32 @@ private String getQualifierPropery(String bundleLocation) {
 
 	//Return a dictionary representing a manifest. The data may result from plugin.xml conversion  
 	private Dictionary<String, String> basicLoadManifest(File bundleLocation) {
-		InputStream manifestStream = null;
-		ZipFile jarFile = null;
+		Hashtable<String, String> result = new Hashtable<>();
 		try {
 			if ("jar".equalsIgnoreCase(IPath.fromOSString(bundleLocation.getName()).getFileExtension()) && bundleLocation.isFile()) { //$NON-NLS-1$
-				jarFile = new ZipFile(bundleLocation, ZipFile.OPEN_READ);
-				ZipEntry manifestEntry = jarFile.getEntry(JarFile.MANIFEST_NAME);
-				if (manifestEntry != null) {
-					manifestStream = jarFile.getInputStream(manifestEntry);
+				try (ZipFile jarFile = new ZipFile(bundleLocation, ZipFile.OPEN_READ)) {
+					ZipEntry manifestEntry = jarFile.getEntry(JarFile.MANIFEST_NAME);
+					if (manifestEntry != null) {
+						try (InputStream manifestStream = jarFile.getInputStream(manifestEntry)) {
```

Comment:
"Closing this ZIP file will close all of the input streams previously returned by invocations of the `getInputStream`".
So I think we can skip the nested try in this case. :)